### PR TITLE
feat: add systemd journal logging configuration

### PR DESCRIPTION
**Pull request by jjaakola-aiven**
# About this change - What it does

Add configurability for logging handlers. The supported handlers are StreamHandler and systemd.journal.JournalHandler.
The log level is preserved for journal as log is written in structured format for journald.

Requires #947 first.

References: #xxxxx

# Why this way

### CODE REVIEW

File: src/karapace/karapace_all.py [hunk 1]
```diff
@@ -21,6 +23,37 @@ class KarapaceAll(KafkaRest, KarapaceSchemaRegistryController):
...
+    log_handler = config.get("log_handler", None)
+    if "systemd" == log_handler:
+        from systemd import journal
+
+        root_handler = journal.JournalHandler(SYSLOG_IDENTIFIER="karapace")
+    if "stdout" == log_handler or log_handler is None:
+        root_handler = logging.StreamHandler(stream=sys.stdout)
+    else:
+        logging.basicConfig(level=logging.INFO, format=log_format)
+        logging.getLogger().setLevel(log_level)
```
Comment (eliax1996) [inline src/karapace/karapace_all.py:40]: maybe add a warn log that the `log_handler` isn't recognized?

File: src/karapace/karapace_all.py [hunk 2]
```diff
@@ -21,6 +23,37 @@ class KarapaceAll(KafkaRest, KarapaceSchemaRegistryController):
...
+    log_level = config.get("log_level", "DEBUG")
+    log_format = config.get("log_format", "%(name)-20s\t%(threadName)s\t%(levelname)-8s\t%(message)s")
+
+    root_handler: logging.Handler | None = None
+    log_handler = config.get("log_handler", None)
+    if "systemd" == log_handler:
+        from systemd import journal
+
+        root_handler = journal.JournalHandler(SYSLOG_IDENTIFIER="karapace")
+    if "stdout" == log_handler or log_handler is None:
```
Comment (eliax1996) [inline src/karapace/karapace_all.py:36]: nit: `elif` to highlight that this is an alternative option other than the first entry?

**Review by eliax1996 — APPROVED** [review]
Except the comments that are optional/non blocking LGTM

File: src/karapace/karapace_all.py (hunk 1 above)
Comment (jjaakola-aiven) [reply src/karapace/karapace_all.py:40]: Done.

File: src/karapace/karapace_all.py (hunk 2 above)
Comment (jjaakola-aiven) [reply src/karapace/karapace_all.py]: Done.

**Review by eliax1996 — APPROVED** [review]

**Pull request merged**